compiler/src/iree/compiler/Codegen/Common/GPU/GPUPruneDPSOps.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/GPU/Passes.h"
+#include "iree/compiler/Codegen/Utils/GPUUtils.h"
+#include "iree/compiler/Dialect/TensorExt/IR/TensorExtOps.h"
+#include "llvm/ADT/STLExtras.h"
+#include "mlir/Dialect/Bufferization/IR/Bufferization.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+
+namespace mlir::iree_compiler {
+
+#define GEN_PASS_DEF_GPUPRUNEDPSOPSPASS
+#include "iree/compiler/Codegen/Common/GPU/Passes.h.inc"
+
+namespace {
+/// Pass to create a `bufferization.alloc_tensor` in private space for all DPS
+/// ops with unused results that can't be removed.
+/// These unused results, if originating from loads from global memory, trigger
+/// allocations in global memory space during bufferization, which will fail.
+/// So, the allocations must be made earlier to avoid failed bufferization.
+/// This pass is a hack to get around other work that can be done to improve
+/// the bufferization algorithm.
+/// For such cases, need to make Flow preserve the unused result as a
+/// result of the dispatch.
+struct GPUPruneDPSOpsPass final
+    : impl::GPUPruneDPSOpsPassBase<GPUPruneDPSOpsPass> {
+  void runOnOperation() override;
+};
+
+void GPUPruneDPSOpsPass::runOnOperation() {
+  MLIRContext *context = &getContext();
+  FunctionOpInterface funcOp = getOperation();
+
+  auto privSpace = gpu::AddressSpaceAttr::get(
+      context, gpu::GPUDialect::getPrivateAddressSpace());
+
+  bufferization::BufferizationOptions options;
+  options.defaultMemorySpaceFn =
+      [&p = privSpace](TensorType t) -> std::optional<Attribute> { return p; };
```

Comment:
I'm -1 on having an alias for the variable. It adds complex to parse the code. If it matters, make it a local function.

```suggestion
      [&privSpace](TensorType t) -> std::optional<Attribute> { return privSpace; };
```